vllm/worker/model_runner.py

```diff
@@ -729,14 +729,42 @@ def add_seq_group(self, seq_group_metadata: SequenceGroupMetadata):
 
     def _use_captured_graph(self,
                             batch_size: int,
+                            decode_only: bool,
                             max_decode_seq_len: int,
                             max_encoder_seq_len: int = 0) -> bool:
-        return (self.decode_only and not self.runner.model_config.enforce_eager
+        return (decode_only and not self.runner.model_config.enforce_eager
                 and batch_size <= _BATCH_SIZES_TO_CAPTURE[-1]
                 and max_decode_seq_len <= self.runner.max_seq_len_to_capture
                 and max_encoder_seq_len <= self.runner.max_seq_len_to_capture
                 and batch_size <= self.runner.max_batchsize_to_capture)
 
+    def _cuda_graph_pad_size(self,
+                             batch_size: int,
+                             num_seqs: int,
+                             max_decode_seq_len: int,
+                             max_encoder_seq_len: int = 0) -> int:
+        is_mscp: bool = self.runner.scheduler_config.is_multi_step and \
+                    self.runner.scheduler_config.chunked_prefill_enabled
+        # The input batch_size is the number of input-tokens that includes
+        # both the prefill and decode tokens. Generally, when the batch has
+        # prefills, we don't use CUDA graphs. i.e. _use_captured_graph() will
+        # be False.
+        # However, In the multi-step + chunked-prefill case, only the first
+        # step has Prefills (if any). The rest of the steps are guaranteed to
+        # be all decodes. In this case, we set up the padding as if all the
+        # sequences are decodes so we may run all steps expect the first step
+        # in CUDA graph mode.
+        batch_size = num_seqs if is_mscp else batch_size
```

Comment:
Is it valid if we just `batch_size = num_seqs` all the time? It seems to me that if you already have `num_seqs`, it's not necessary to take batch_size in this function.

Reply:
It is valid in the case when all the input sequences are decodes. batch-size in the rest of the code refers to the number of input tokens being batched. I did not want to conflate the two ideas.

However, I have made some changes to this function, that simplifies the interface a little and makes this conflation  legitimate. Please take a look. Appreciate any feedback on this! Thanks.